### PR TITLE
[WIP] List workers

### PR DESCRIPTION
**Pull request by nyrosmith**
Associated Issue: #3814 

This is a WIP and the code is quite naive now.

* get the debuggee's workers
* set them in the redux store
* show the workers in the Workers panel

### CODE REVIEW

File: src/components/SecondaryPanes/Workers.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.workers-list * {
+  -moz-user-select: none;
+  user-select: none;
+}
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/Workers.css]: We use autoprefix so you don't need the moz prefix

File: src/reducers/debuggee.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const initialState = {
```
Comment (jasonLaster) [inline src/reducers/debuggee.js]: Let's add flow '// @flow'

File: src/components/SecondaryPanes/Workers.css (hunk 1 above)
Comment (nyrosmith) [reply src/components/SecondaryPanes/Workers.css]: I looked how breakpoints are styled so I'll remove it there too

**Comment by nyrosmith** [comment]
Is this failing test really caused by my changes?

**Comment by jasonLaster** [comment]
nope, those tests are not you. They're two known intermittents

**Comment by jasonLaster** [comment]
@nyrosmith i pushed some extra boilerplate for the reducers

* immutable
* flow types
* reselect

The patch is not done, it is just a start. Happy to help you finish, but thought it would be nice to leave it as an exercise so you can learn the architecture. some of this probably seems like a bit much for a small reducer, but it's helpful as it scales.

File: src/client/firefox.js [hunk 3]
```diff
@@ -35,6 +35,10 @@ export async function onConnect(connection: any, actions: Object) {
     wasmBinarySource: supportsWasm
   });
 
+  threadClient._parent.listWorkers().then(workers =>
```
Comment (bomsy) [inline src/client/firefox.js]: @nyrosmith `threadClient._parent` can be undefined, you probably need to check if it exists before you call `listWorkers`

**Comment by nyrosmith** [comment]
Everything green :-)

giphy-downsized

**Pull request merged**